internal/fvalidate/verapdf_validator.go

```diff
@@ -0,0 +1,76 @@
+package fvalidate
+
+import (
+	"errors"
+	"os/exec"
+
+	"github.com/go-logr/logr"
+)
+
+// pdfaPUIDs are the https://www.nationalarchives.gov.uk/pronom/ IDs of the
+// PDF/A formats.
+var pdfaPUIDs = []string{
+	"fmt/95",   // PDF/A 1a
+	"fmt/354",  // PDF/A 1b
+	"fmt/476",  // PDF/A 2a
+	"fmt/477",  // PDF/A 2b
+	"fmt/478",  // PDF/A 2u
+	"fmt/479",  // PDF/A 3a
+	"fmt/480",  // PDF/A 3b
+	"fmt/481",  // PDF/A 3u
+	"fmt/1910", // PDF/A 4
+	"fmt/1911", // PDF/A 4e
+	"fmt/1912", // PDF/A 4f
+}
+
+type veraPDFValidator struct {
+	cmd    string
+	logger logr.Logger
+}
+
+var _ Validator = (*veraPDFValidator)(nil)
+
+func NewVeraPDFValidator(cmd string, logger logr.Logger) *veraPDFValidator {
+	return &veraPDFValidator{cmd: cmd, logger: logger}
+}
+
+func (v *veraPDFValidator) FormatIDs() []string {
+	return pdfaPUIDs
+}
+
+func (v *veraPDFValidator) Name() string {
+	return "veraPDF"
+}
+
+func (v *veraPDFValidator) Validate(path string) (string, error) {
+	// If the veraPDF cmd path is not set then skip validation.
+	if v.cmd == "" {
+		return "", nil
+	}
+
+	cmd := exec.Command(v.cmd, "--recurse", path) // #nosec: G204 -- trusted path.
+	_, err := cmd.Output()
+	if err != nil {
+		switch e := err.(type) {
```

Comment:
We are only considering one case in this switch, I think it would be easier to read as part of the same if. The default case is the same as the else.

Reply:
All four of the returns in this logic tree are different, maybe this makes the logic branches clearer?

```
	if err != nil {
		if e, ok := err.(*exec.ExitError); ok {
			switch code := e.ExitCode(); code {
			case 1:
				// Exit code 1 indicates a validation error, and there is no
				// STDERR. In this case the full validation report is written to
				// STDOUT, but we are ignoring it right now because it is very
				// long.
				return "One or more PDF/A files are invalid", nil

			default:
				// Other exit codes (e.g. file not found) should write an error
				// message to STDERR.
				v.logger.Info("veraPDF validate", "exit code", e.ExitCode(), "STDERR", string(e.Stderr))
				return "", errors.New("PDF/A validation failed with an application error")
			}
		}
		return "", err
	}

	return "", nil
```

Reply:
I couldn't find any formal rule or guide in our developer resources but I think our practice is to usually return an explicit `nil` error from a function on the "happy" branch where no errors are encountered.  I think in a case like this returning `return "", err` where `err` can be nil or have an error value makes it harder to distinguish when an error is returned and when it isn't.

In this particular case, it might actually be better to return early if `err` is nil to avoid unnecessary nesting:

```
if err == nil { // error IS nil.
   return "", nil
}

if e, ok := err.(*exec.ExitError); !ok {
	return "", err
}

switch code := e.ExitCode(); code {
case 1:
	// Exit code 1 indicates a validation error, and there is no
	// STDERR. In this case the full validation report is written to
	// STDOUT, but we are ignoring it right now because it is very
	// long.
	return "One or more PDF/A files are invalid", nil

default:
	// Other exit codes (e.g. file not found) should write an error
	// message to STDERR.
	v.logger.Info("veraPDF validate", "exit code", e.ExitCode(), "STDERR", string(e.Stderr))
	return "", errors.New("PDF/A validation failed with an application error")
}
```

Reply:
Now seeing this, the second switch also seems unnecessary and could be solved with an `if`. Not a big deal, whatever you prefer, I just found the two nested switches with a single case a bit too much.

Reply:
I like the switch because it reads well to me and makes it easy to add more exit codes if we find that necessary in the future.  I can change it to and "if" conditional if you feel strongly about it.